### ISSUE TITLE
No negative x_scores

### ISSUE THREAD

**Issue by sdittmer** [wangxin0716/ssl#6]
I am unsure whether [this line](https://github.com/p3i0t/SimCLR-CIFAR10/blob/2f449c2e39666a5c3439859347e3f1aced67b17d/simclr.py#L54): `x_scores =  (x @ x.t()).clamp(min=1e-7)`, is a correct implementation of SimCLR. I think the `clamp` should not be there. It might make sense to use it, but it changes the SimCLR idea from "_for negative examples, we want projections that point into opposite directions_" to "_for negative examples, we want projections that are at least orthogonal._"

**Comment by dkobak** [comment]
I agree. The clamping is not there in the SimCLR paper. Cosine similarity can be negative.
